cartoframes/context.py

```diff
@@ -329,98 +332,32 @@ def write(self, df, table_name, temp_dir=CACHE_DIR, overwrite=False,
                   similar arguments.
 
         Returns:
-            :py:class:`BatchJobStatus <cartoframes.batch.BatchJobStatus>` or
-            None: If `lnglat` flag is set and the DataFrame has more than
-            100,000 rows, a :py:class:`BatchJobStatus
-            <cartoframes.batch.BatchJobStatus>` instance is returned.
-            Otherwise, None.
+            :py:class:`Dataset <cartoframes.datasets.Dataset>`
 
         .. note::
             DataFrame indexes are changed to ordinary columns. CARTO creates
             an index called `cartodb_id` for every table that runs from 1 to
             the length of the DataFrame.
         """  # noqa
         # work on a copy to avoid changing the original
-        _df = df.copy()
-        if not os.path.exists(temp_dir):
-            self._debug_print(temp_dir='creating directory at ' + temp_dir)
-            os.makedirs(temp_dir)
-        if encode_geom:
-            _add_encoded_geom(_df, geom_col)
-
-        if not overwrite:
-            # error if table exists and user does not want to overwrite
-            self._table_exists(table_name)
-        elif kwargs.get('privacy') is None:
-            # get privacy so it's not overwritten on write
-            privacy = self._get_privacy(table_name)
-            if privacy:
-                kwargs['privacy'] = privacy
-
-        # issue warning if the index is anything but the pandas default
-        #  range index
-        if not _df.index.equals(pd.RangeIndex(0, _df.shape[0], 1)):
-            _df.reset_index(inplace=True)
-
-        pgcolnames = utils.normalize_colnames(_df.columns)
-        if table_name != utils.norm_colname(table_name):
-            table_name = utils.norm_colname(table_name)
-            warn('Table will be named `{}`'.format(table_name))
-
-        if _df.shape[0] > MAX_IMPORT_ROWS:
-            # NOTE: schema is set using different method than in _set_schema
-            # send placeholder table
-            final_table_name = self._send_dataframe(_df.iloc[0:0], table_name,
-                                                    temp_dir, geom_col,
-                                                    pgcolnames, kwargs)
-            # send dataframe in batches, combine into placeholder table
-            final_table_name = self._send_batches(_df, table_name, temp_dir,
-                                                  geom_col, pgcolnames, kwargs)
-        else:
-            final_table_name = self._send_dataframe(_df, table_name, temp_dir,
-                                                    geom_col, pgcolnames,
-                                                    kwargs)
-            self._set_schema(_df, final_table_name, pgcolnames)
+        tqdm.write('Params: encode_geom, geom_col and everything in kwargs are deprecated and not being used any more')
+        dataset = Dataset(self, table_name, df=df)
```

Comment:
What about a classmethod like:
```python
Dataset(self, table_name).from_dataframe(df)
```
Or some variation of that? I'm just throwing out some other ideas.